src/AddIns/Uno.UI.Lottie/LottieVisualSource.Skottie.cs

```diff
@@ -0,0 +1,388 @@
+﻿#nullable enable
+
+#if __SKIA__ || NET6_0
+using System;
+using System.Threading;
+using Windows.Foundation;
+using Windows.UI.Xaml.Controls;
+using System.Threading.Tasks;
+using Uno.Disposables;
+using SkiaSharp;
+using Uno.Foundation.Logging;
+using Windows.UI.Xaml;
+using Windows.System;
+using System.Diagnostics;
+using SkiaSharp.SceneGraph;
+using Windows.UI.Xaml.Media;
+
+#if HAS_UNO_WINUI
+using SkiaSharp.Views.Windows;
+#else
+using SkiaSharp.Views.UWP;
+#endif
+
+namespace Microsoft.Toolkit.Uwp.UI.Lottie
+{
+	partial class LottieVisualSourceBase
+	{
+		private UIElement? _renderSurface;
+		private SkiaSharp.Skottie.Animation? _animation;
+		private SKXamlCanvas? _softwareCanvas;
+		private SKSwapChainPanel? _hardwareCanvas;
+		private DispatcherQueueTimer? _timer;
+
+		public bool UseHardwareAcceleration { get; set; }
+#if __SKIA__
+			// SkiaSharp.Views.Uno uses the underlying canvas for hardware acceleration.
+			= false;
+#else
+			= true;
+#endif
+
+		private Uri? _lastSource;
+		private (double fromProgress, double toProgress, bool looped)? _playState;
+
+		private Stopwatch _stopwatch = new Stopwatch();
+		private TimeSpan? _progress;
+
+		private InvalidationController? _invalidationController;
+		private readonly SerialDisposable _animationDataSubscription = new SerialDisposable();
+
+		async Task InnerUpdate(CancellationToken ct)
+		{
+			var player = _player;
+
+			if (player == null)
+			{
+				return;
+			}
+
+			await SetProperties();
+
+			async Task SetProperties()
+			{
+				var sourceUri = UriSource;
+				if (_lastSource == null || !_lastSource.Equals(sourceUri))
+				{
+					_lastSource = sourceUri;
+					if ((await TryLoadDownloadJson(sourceUri, ct)) is { } jsonStream)
+					{
+						var cacheKey = sourceUri.OriginalString;
+						_animationDataSubscription.Disposable = null;
+						_animationDataSubscription.Disposable =
+							LoadAndObserveAnimationData(jsonStream, cacheKey, OnJsonChanged);
+
+						void OnJsonChanged(string updatedJson, string updatedCacheKey)
+						{
+							try
+							{
+								if (SkiaSharp.Skottie.Animation.TryParse(updatedJson, out var animation))
+								{
+									animation.Seek(0);
+
+									if (this.Log().IsEnabled(LogLevel.Debug))
+									{
+										this.Log().Debug($"Version: {animation.Version} Duration: {animation.Duration} Fps:{animation.Fps} InPoint: {animation.InPoint} OutPoint: {animation.OutPoint}");
+									}
+								}
+								else
+								{
+									throw new InvalidOperationException("Failed to load animation.");
+								}
+
+								SetAnimation(animation);
+
+								if (_playState != null)
+								{
+									var (fromProgress, toProgress, looped) = _playState.Value;
+									Play(fromProgress, toProgress, looped);
+								}
+							}
+							catch(Exception ex)
+							{
+								throw new Exception("", ex);
```

Comment:
why ? that will lose the original stack trace

Reply:
It's a test, but currently, I'm having trouble making SkiaSharp run properly on iOS. I'm mostly validating the packaging with this draft :)